record/record_test.go

```diff
@@ -326,30 +341,69 @@ func writeRecords(records [][]byte) ([]byte, error) {
 			return nil, err
 		}
 	}
+
 	if err := w.Close(); err != nil {
 		return nil, err
 	}
-	return buf.Bytes(), nil
+
+	results.buf = buf.Bytes()
+	return results, nil
 }
 
-func TestBasicRecover(t *testing.T) {
-	records := [][]byte{
-		[]byte(strings.Repeat("a", blockSize-headerSize)),
-		[]byte(strings.Repeat("b", blockSize-headerSize)),
-		[]byte(strings.Repeat("c", blockSize-headerSize)),
+// Corrupt the checksum of the record that starts the the specified block
```

Comment:
Doc comments should start with the name of the thing being commented upon. Thus:
// corruptBlock corrupts the checksum of the etc.